pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -71,20 +75,44 @@ func getDefaultCircuitBreakerThresholds() *cluster.CircuitBreakers_Thresholds {
 // For outbound: Cluster for each service/subset hostname or cidr with SNI set to service hostname
 // Cluster type based on resolution
 // For inbound (sidecar only): Cluster for each inbound endpoint port and for each service port
-func (configgen *ConfigGeneratorImpl) BuildClusters(proxy *model.Proxy, push *model.PushContext) ([]*discovery.Resource, model.XdsLogDetails) {
+// if attemptDeltas is set to true, then BuildClusters will attempt to build delta clusters for that proxy
+func (configgen *ConfigGeneratorImpl) BuildClusters(proxy *model.Proxy, push *model.PushContext, attemptDeltas bool, updates *model.PushRequest, watched *model.WatchedResource) ([]*discovery.Resource, model.XdsLogDetails) {
```

Comment:
`watched` is not used.

Do we need `attemptDeltas` and `updates`? Maybe we can just make non-nil `updates` meant to use delta